src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -59,9 +59,13 @@ private List<BibEntry> getBibEntries(URL urlForQuery) throws FetcherException {
             fetchedEntries.forEach(this::doPostCleanup);
             return fetchedEntries;
         } catch (IOException e) {
-            throw new FetcherException("A network error occurred while fetching from " + urlForQuery, e);
+            throw new FetcherException(("A network error occurred while fetching from " + urlForQuery).replaceAll
+//                   // Regular expression to redact API keys from the error message
+                    ("(?i)(api[_-]?key)=.*","[REDACTED]"), e);
```

Comment:
move this regex to a static Pattern with Pattern.compile
and then later you use  YOUR_PATTERN.matcher(str).replaceAll(repl)

Reply:
done!